modules/network/src/main/java/org/apache/ignite/network/internal/netty/ConnectionManager.java

```diff
@@ -110,21 +125,24 @@ public SocketAddress getLocalAddress() {
 
     /**
      * Gets a {@link NettySender}, that sends data from this node to another node with the specified address.
+     * @param consistentId Another node's consistent id.
      * @param address Another node's address.
      * @return Sender.
      */
-    public CompletableFuture<NettySender> channel(SocketAddress address) {
-        NettySender channel = channels.compute(
-            address,
-            (addr, sender) -> (sender == null || !sender.isOpen()) ? null : sender
-        );
-
-        if (channel != null)
-            return CompletableFuture.completedFuture(channel);
+    public CompletableFuture<NettySender> channel(@Nullable String consistentId, SocketAddress address) {
+        if (consistentId != null) {
```

Comment:
Can you add some comments in the code explaining the logic of this method? I already forgot the correspondence between `channels` and `clients` map =(

Reply:
Sure